DatadogRUM/Sources/RUMMonitor/Scopes/RUMApplicationScope.swift

```diff
@@ -85,75 +93,118 @@ internal class RUMApplicationScope: RUMScope, RUMContextProvider {
+            startPrecondition = .userAppLaunch
+        }
 
-    private func createInitialSession(on command: RUMCommand, context: DatadogContext, writer: Writer) {
         let initialSession = RUMSessionScope(
             isInitialSession: true,
             parent: self,
             startTime: context.sdkInitDate,
+            startPrecondition: startPrecondition,
             dependencies: dependencies,
             hasReplay: context.hasReplay
         )
 
+        lastSessionEndReason = nil
         sessionScopes.append(initialSession)
         sessionScopeDidUpdate(initialSession)
     }
 
-    private func applicationStart(on command: RUMCommand, context: DatadogContext, writer: Writer) {
-        applicationActive = true
-
-        guard context.applicationStateHistory.currentSnapshot.state != .background else {
-            return
+    /// Starts new RUM Session immediately after previous one expires or time outs. It transfers some of the state from the expired session to the new one.
+    private func refresh(expiredSession: RUMSessionScope, on command: RUMCommand, context: DatadogContext, writer: Writer) -> RUMSessionScope {
+        var startPrecondition: RUMSessionPrecondition? = nil
+
+        if lastSessionEndReason == .timeOut {
+            startPrecondition = .inactivityTimeout
+        } else if lastSessionEndReason == .maxDuration {
+            startPrecondition = .maxDuration
+        } else {
+            dependencies.telemetry.error("Failed to determine session precondition for REFRESHED session with end reason: \(lastSessionEndReason?.rawValue ?? "unknown"))")
         }
 
-        // Immediately start the ApplicationLaunchView for the new session
-        _ = process(
-            command: RUMApplicationStartCommand(
-                time: command.time,
-                attributes: command.attributes
-            ),
-            context: context,
-            writer: writer
+        let refreshedSession = RUMSessionScope(
+            from: expiredSession,
+            startTime: command.time,
+            startPrecondition: startPrecondition,
+            context: context
         )
+        sessionScopeDidUpdate(refreshedSession)
+        lastSessionEndReason = nil
+        _ = refreshedSession.process(command: command, context: context, writer: writer)
+        return refreshedSession
     }
 
+    /// Starts new RUM Session some time after previous one was ended with ``RUMMonitorProtocol.stopSession()`` API. It may re-activate the last view from previous session.
     private func startNewSession(on command: RUMCommand, context: DatadogContext, writer: Writer) {
+        var startPrecondition: RUMSessionPrecondition? = nil
+
+        if lastSessionEndReason == .stopAPI {
+            startPrecondition = .explicitStop
+        } else {
+            dependencies.telemetry.error("Failed to determine session precondition for NEW session with end reason: \(lastSessionEndReason?.rawValue ?? "unknown"))")
+        }
```

Comment:
/question The condition for starting a new session is `activeSession == nil && command.isUserInteraction`, so shouldn't we set the precondition to `.userAppLaunch` by default instead?

Reply:
This is one of the tricks in the current implementation. This flow is reached due to `stopSession()` API, as proven in [this test](https://github.com/DataDog/dd-sdk-ios/blob/8aab1db2b7043435515335d45c5188825665396a/DatadogRUM/Tests/RUMMonitor/Scopes/RUMApplicationScopeTests.swift#L506-L528). The condition has two sides and while the rhs means user interaction, the lhs guards "there is no more active session". Because we start initial session with SDK start (https://github.com/DataDog/dd-sdk-ios/pull/1278) this can be effectively reached only for later sessions.

But you touched something with this question - we assume "this is not initial session". I will add telemetry to prove if this assumption is wrong.